### DISCUSSION TITLE
How i can add regex filter in if construction?

### DISCUSSION THREAD

**Discussion by jidckii** [goreleaser/goreleaser#4023] [Q&A] [answered]
I want to upload postfix releases to a special directory, but I can't figure out the right condition for this. Can you suggest how to do it?

I tryed this 

```yaml
folder: '{{- if .IsNightly }}nightly/{{.Version}}{{- else if filter .Version "v\d+\.\d+\.\d+-(dev|alpha|beta|rc)\d*" }}beta/{{.Version}}{{else}}{{.Version}}{{- end }}'
```

 but, has error 

```
error=template: tmpl:1: invalid syntax
```

**Comment by caarlos0** [comment, marked as answer]
Hey, there is no `filter` function, but in this case you can simply check if `.Prerelease` field is set, e.g.:

```yaml
folder: '{{- if .IsNightly }}nightly/{{.Version}}{{- else if .PreRelease }}beta/{{.Version}}{{else}}{{.Version}}{{- end }}'
```

**Reply by caarlos0** [reply]
correction: there is the a `filter` function indeed, I forgot about it... not sure why its not working in your case, though, will check.

Original answer should still be a valid/simpler alternative though

**Reply by caarlos0** [reply]
it looks like your regex is not valid, not sure why though, maybe needs more escaping due to templates...
